src/components/File.jsx

```diff
@@ -94,6 +98,20 @@ class File extends Component {
     )
   }
 
+  renderFilenameInput (attributes) {
```

Comment:
In other words, there is no reason that the File Component embeds functions that are not related to the File Component as *display a input field* or *display a *rich filename cell*.